### DISCUSSION TITLE
What branch is version 7.4 building from?

### DISCUSSION THREAD

**Discussion by RichardMeadowsTC** [PowerShell/PowerShell#19090] [Q&A] [answered]
I built main branch last night.  Got version 7.3.0- preivew 3.  I was not able to find any branch for 7.4.0-preview1.

**Comment by rhubarb-geek-nz** [comment, marked as answer]
Try tag [v7.4.0-preview.1](https://github.com/PowerShell/PowerShell/tree/v7.4.0-preview.1)

**Comment by iSazonov** [comment]
> I built main branch last night. Got version 7.3.0- preivew 3. I was not able to find any branch for 7.4.0-preview1.

"7.3.0- preivew 3" is last annotated tag. Latest ones are nonannotated tags and build script can not find them.

Really master branch is latest 7.4.0.

**Reply by kborowinski** [reply]
@RichardMeadowsTC Exactly what @iSazonov said. You can add your own annotated tag to your local repo, like I do:
![image](https://user-images.githubusercontent.com/16168755/216607818-227fcd31-2c82-4228-99ce-14e3473d1b28.png)

**Reply by RichardMeadowsTC** [reply]
> > I built main branch last night. Got version 7.3.0- preivew 3. I was not able to find any branch for 7.4.0-preview1.
> 
> "7.3.0- preivew 3" is last annotated tag. Latest ones are nonannotated tags and build script can not find them.
> 
> Really master branch is latest 7.4.0.

What does the build script do when it is building to get actual last official tag?
How / where is new work added?

**Reply by kborowinski** [reply]
It basically calls `git describe` to get the latest annotated tag:
![image](https://user-images.githubusercontent.com/16168755/216633322-b6f9fade-42f9-4cf3-9261-db110949d3ed.png)

And since the `v7.3.0-preview.3` was last *annotated* tag on `master` repository, all local builds of PowerShell will have it for version number.

`Master` is the branch where new work is added. Whenever team decides to release preview version they create a new branch for it.

**Reply by RichardMeadowsTC** [reply]
Who makes the tags, and from what branch.  As there isn't a 7.4.x branch anywhere.

**Reply by jborean93** [reply]
Master is 7.4. Fetch the latest tags or add your own tag to have a different version be reported when you build powershell.

**Reply by RichardMeadowsTC** [reply]
I just pulled latest from master.  Git describe says v7.3.0-preview.3-610-gec0eb2206.  That is not 7.4

**Reply by jborean93** [reply]
I don't believe they've updated the tags yet on that branch but master is the main one that 7.4 is based off of. You can even see the commits for master are quick recent https://github.com/PowerShell/PowerShell/commits/master. Once 7.4 is starting the normal release/rc process there will be a new branch made. For now it's just based on what is in master. Notice how the `7.4.0-preview.1` tag only goes up to when it was actually released https://github.com/PowerShell/PowerShell/commits/v7.4.0-preview.1. The 7.4 preview contains more recent commits than that in the `master` branch.

**Reply by kborowinski** [reply]
From what I understood, the team will not update tags on `master` anymore. I've created few issues about it #16363, #15244, #14439 and the explanation is [that merging tags back would cause too many conflicts](https://github.com/PowerShell/PowerShell/issues/16363#issuecomment-959759234).

**Reply by iSazonov** [reply]
I guess there are many strategies for the release process, but if it is based on the release branch today, I don't see where the conflicts can come from. Moreover, it is unclear that if the tags are created anyway, but as lightweight, why they can not be made annotated.

**Reply by RichardMeadowsTC** [reply]
The entire process is very much hidden.  
"https://github.com/PowerShell/PowerShell/commits/v7.4.0-preview.1. The 7.4 preview contains more recent commits than that in the master branch."  Yet there is no branch for 7.4 that I can see.

There seams to be a small group managing the project.  Not at all open.

**Reply by kborowinski** [reply]
That's not completely right. Last commit in *7.4.0-preview.1* is from Dec 20:
![image](https://user-images.githubusercontent.com/16168755/217302335-cbb4a092-ebd6-472d-ab77-ba5a0d4da193.png)

and in `master`, just 4 days ago:
![image](https://user-images.githubusercontent.com/16168755/217302549-e8481719-e4e9-4d5a-a9d9-836a26117edb.png)

There are **73** new commits in `master` since *7.4.0-preview.1* release:
![image](https://user-images.githubusercontent.com/16168755/217303426-fd97eeb8-561b-4ab5-8f67-ac8349165b3d.png)

@iSazonov Anything that would help improve the release process would be great.

**Reply by iSazonov** [reply]
> There seams to be a small group managing the project. Not at all open.

@RichardMeadowsTC Release process is internal MSFT process but all commits are open. A problem is rather that they are not very concerned about alternative release processes that exist in community.

> @iSazonov Anything that would help improve the release process would be great.

@kborowinski I've been trying to talk them into it for years, just like you. No doubt the PowerShell MSFT team has an understanding that the community needs to be supported (although their understanding may be different than ours). However, public pressure will not work because they are under corporate pressure, which is a higher priority for them. In my experience, other channels work much more effectively, like MSFT partners, MVPs and others when there is direct exposure to departments that can trigger powerful internal processes.
We can only hope that they will at least explain how to work with this repository if the tags seem to be, but now it seems that they are not, although they are all there but in a different form. This is chaos.
